File: 2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.cpp
class Solution {
public:
    //code by vishu badmos
    string largestGoodInteger(string num) {
        string str = "";
        for(int i=0;i<=num.length()-3;i++){
            string temp = num.substr(i,3); // take a substring of length three
            if(temp[0]==temp[1] && temp[0]==temp[2]){
                str = max(str,temp);
            }
        }
        return str;
    }
};

File: 2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.js
/**
 * @param {string} num
 * @return {string}
 */
var largestGoodInteger = function(num) {
    //so i have a string function 
    let str = new String(num); // take the string in other 
    let size = str.length // this will store the size our string
    let ans = "";
    for(let i=0;i<=size-3;i++){
        let temp = str.substr(i,3);
        console.log(temp); // store the three characters in the temp 
        if(temp[0]==temp[1] && temp[0]==temp[2]){
            if(temp>ans){
                ans = temp;
            }
        }
    }
    return ans;
};